resources/splunk_app.rb

```diff
@@ -162,30 +163,40 @@ class PackagedApp < SplunkApp
 
   action :install do
     return unless !version || changed?(:version)
+    @converge_actions = nil # Clear resource updates by property change
 
     package_path = app_cache_path + CernerSplunk::PathHelpers.filename_from_url(source_url).gsub(/.spl$/, '.tgz')
 
     remote_file package_path.to_s do
       source source_url
-      show_progress true if defined? show_progress # Chef 12.9 feature
+      show_progress true
     end
 
-    directory app_path.to_s do
+    directory 'ensure app path exists' do
+      path app_path.to_s
+      recursive true
       action :create
     end
 
     backup_app if changed? :version
 
-    poise_archive package_path.to_s do
-      destination((app_cache_path + 'new').to_s)
+    temp_path = (app_cache_path + 'new').to_s
```

Comment:
I'm curious, what issue did moving this out to a separate variable declaration fix?